src/H5Dbtree2.c

```diff
@@ -1003,16 +1076,36 @@ H5D__bt2_idx_get_addr(const H5D_chk_idx_info_t *idx_info, H5D_chunk_ud_t *udata)
 } /* H5D__bt2_idx_get_addr() */
 
 /*-------------------------------------------------------------------------
- * Function:	H5D__bt2_idx_iterate_cb
+ * Function:    H5D__bt2_idx_load_metadata
+ *
+ * Purpose:     Load additional chunk index metadata beyond the chunk index
+ *              itself. Currently a no-op.
  *
- * Purpose:	Translate the B-tree specific chunk record into a generic
+ * Return:      Non-negative on success/Negative on failure
+ *
+ *-------------------------------------------------------------------------
+ */
+static herr_t
+H5D__bt2_idx_load_metadata(const H5D_chk_idx_info_t H5_ATTR_UNUSED *idx_info)
```

Comment:
Same comment about loading root note

Reply:
Hmm, yes looks like I should probably add a lookup here for the root node.

Reply:
It might take too much time to get in for 1.14.3 since I think you'd need to add a H5B2 function, but something to keep in mind for later

Reply:
It looks like performing a bt2 `get_addr` call will have the same effect as earray and farray where in this case a fake chunk lookup will make sure the root node is read in collectively. We should eventually update the interfaces behind the chunk indices to have a more explicit way of controlling this in the future though.

Reply:
This will also open other some internal nodes and the leaf node for chunk 0, which is honestly fine.

Reply:
Though since there's multiple steps (need to read each node to get the address of the node further down) it could be wasted overhead if no process actually needs any data on that "side" of the dataset. But again, something to keep in mind for later.

Reply:
Taking this further, we could, after collectively loading only the root node, collectively load every node at the next level down in a single operation, then again at the next level, and so on (up to a limit that maybe could be set depending on the size of the metadata cache and the average node size)

Reply:
To prime the metadata cache in the most efficient way possible

Reply:
Agreed that without having explicit control over loading the root node, there may be some wasted effort in loading the data for that chunk if no one needs it. I suppose the hope is that if the user has collective metadata reads enabled, the benefit of having the root node loaded at scale will be more worth it. That and the ranks that are interested in the chunk won't end up reading it independently.